### PR TITLE
Update commit id in rpm specs when auto updating yamls

### PR DESCRIPTION
**Pull request by alanoe**

### CODE REVIEW

File: lib/rpm_package.py [hunk 1]
```diff
@@ -119,6 +119,32 @@ def update_prerelease_tag(self, new_prerelease):
         LOG.info("Updated '%s' prerelease tag to: %s"
                  % (self.path, new_prerelease))
 
+    def update_commit_id(self, old_commit_id, new_commit_id):
+
+        # change log may contain old commit IDs and we do not want to replace them
+
+        # read up to change log
+        lines = []
+        change_log_tag = "%changelog"
```
Comment (ghost) [inline lib/rpm_package.py]: This is a constant. Use uppercase.

File: tools/upgrade_versions.py [hunk 2]
```diff
@@ -26,6 +26,7 @@
 from lib import packages_manager
 from lib import repository
 from lib import rpm_package
+from lib.utils import replace_str_in_file
```
Comment (ghost) [inline tools/upgrade_versions.py:29]: This move is not mentioned in the commit message and could be separated to another commit.

File: tools/upgrade_versions.py [hunk 3]
```diff
@@ -104,9 +104,9 @@ def update(self, user_name, user_email):
                 (self.pkg.version, self._repo_version))
 
         pkg.spec_file.update_prerelease_tag(self._repo_prerelease)
-        self._bump_release(pkg, change_log_header, user_name, user_email)
+        self._update_release_and_change_log(pkg, change_log_header, user_name, user_email)
 
-    def _bump_release(self, pkg, change_log_header=None, user_name=None,
+    def _update_release_and_change_log(self, pkg, change_log_header=None, user_name=None,
                       user_email=None):
```
Comment (ghost) [inline tools/upgrade_versions.py]: Indentation is not following PEP8.

File: tools/upgrade_versions.py [hunk 4]
```diff
@@ -128,7 +128,7 @@ def _bump_release(self, pkg, change_log_header=None, user_name=None,
         if change_log_lines:
             assert user_name is not None
             assert user_email is not None
-            pkg.spec_file.bump_release(change_log_lines, user_name, user_email)
+            pkg.spec_file.update_release_and_change_log(change_log_lines, user_name, user_email)
```
Comment (ghost) [inline tools/upgrade_versions.py]: Line is too long.

File: tools/upgrade_versions.py [hunk 5]
```diff
@@ -173,35 +173,36 @@ def _read_version_from_repo(self, repo_path):
             raise exception.PackageError(msg)
 
 
-def push_new_versions(versions_repo, release_date, versions_repo_push_url,
-                      versions_repo_push_branch, committer_name,
-                      committer_email):
+def commit_and_push_new_versions(versions_repo, release_date, versions_repo_push_url,
+                      versions_repo_push_branch, updater_name,
+                      updater_email, push):
```
Comment (ghost) [inline tools/upgrade_versions.py]: Indentation is not following PEP8.

File: lib/config.py [hunk 6]
```diff
@@ -62,15 +62,19 @@
...
         dict(help='URL of the repository used for pushing'),
     ('--push-repo-branch',):
         dict(help='Branch of the repository used for pushing',
              default='master'),
-    ('--committer-name',):
-        dict(help='Name used when creating a commit and bumping spec files'),
-    ('--committer-email',):
-        dict(help='Email used when creating a commit and bumping spec files'),
+    ('--updater-name',):
+        dict(help='Name used when updating files'),
```
Comment (ghost) [inline lib/config.py]: I think it's better to specify what we are updating.
For example: `Name used when updating spec files and committing changes`
or `Name used to update spec files changelogs and create git commit`

File: tools/upgrade_versions.py [hunk 7]
```diff
@@ -232,9 +238,11 @@ def run(CONF):
...
         pkg.unlock()
 
-    release_date = datetime.today().date().isoformat()
-    push_new_versions(versions_repo, release_date, push_repo_url,
-                      push_repo_branch, committer_name, committer_email)
+    if commit_updates:
+        release_date = datetime.today().date().isoformat()
+        commit_and_push_new_versions(versions_repo, release_date, push_repo_url,
+                                     push_repo_branch, updater_name, updater_email,
+                                     push_updates)
```
Comment (ghost) [inline tools/upgrade_versions.py]: I think it would be better to divide this function in two to reduce the number of parameters. I suggest:
`commit_new_versions(versions_repo, release_date, updater_name, updater_email)`
`push_head_commit(versions_repo, push_repo_url, push_repo_branch)`

**Review by ghost — CHANGES_REQUESTED** [review]

File: lib/config.py [hunk 8]
```diff
@@ -62,15 +62,19 @@
         dict(help='Branch of the release notes repository to checkout'),
 }
 PUSH_REPO_ARGS = {
+    ('--commit-updates',):
+        dict(help='Commit file updates to local repository'),
+    ('--push-updates',):
+        dict(help='Push file updates to remote repository'),
```
Comment (ghost) [inline lib/config.py]: Those parameters will require a string argument, and the resulting variable's value will always evaluate to `True`.
Since the default we want is the boolean value `True`, and we only need the parameters to disable those actions, we should use `action='store_false'` and rename the parameters to something like `--no-commit-updates` and `--no-push-updates`. We also need to pass `dest=commit_updates` and `dest=push_updates`.
If you really wanted to keep the parameter names, you could pass `type=ast.literal_eval`, so `--commit-updates False` would set the variable to the bolean `False`, but I don't think this is a good approach.

File: lib/config.py (hunk 6 above)
Comment (alanoe) [reply lib/config.py]: Agreed. Done

File: tools/upgrade_versions.py (hunk 7 above)
Comment (alanoe) [reply tools/upgrade_versions.py]: I prefer to do this split in a separate patch then. Done

File: tools/upgrade_versions.py (hunk 5 above)
Comment (alanoe) [reply tools/upgrade_versions.py]: It was not already following it before. I'll use the opportunity to fix it anyway.

File: tools/upgrade_versions.py (hunk 3 above)
Comment (alanoe) [reply tools/upgrade_versions.py]: Done

File: tools/upgrade_versions.py (hunk 2 above)
Comment (alanoe) [reply tools/upgrade_versions.py:29]: Done

File: lib/rpm_package.py (hunk 1 above)
Comment (alanoe) [reply lib/rpm_package.py]: Done

File: lib/config.py (hunk 8 above)
Comment (alanoe) [reply lib/config.py]: Done

File: tools/upgrade_versions.py (hunk 4 above)
Comment (alanoe) [reply tools/upgrade_versions.py]: Done

File: tools/upgrade_versions.py (hunk 2 above)
Comment (ghost) [reply tools/upgrade_versions.py:29]: This does not belong to this patch. It is duplicated.

File: tools/upgrade_versions.py [hunk 9]
```diff
@@ -229,13 +229,18 @@ def run(CONF):
...
-    REQUIRED_PARAMETERS = ["push_repo_url", "push_repo_branch",
-                           "committer_name", "committer_email"]
+    updater_name = CONF.get('default').get('updater_name')
+    updater_email = CONF.get('default').get('updater_email')
+
+    if push_updates:
+        REQUIRED_PARAMETERS = ["push_repo_url", "push_repo_branch"]
+    else:
+        REQUIRED_PARAMETERS = []
+    REQUIRED_PARAMETERS += ["updater_name", "updater_email"]
```
Comment (ghost) [inline tools/upgrade_versions.py]: You could set `REQUIRED_PARAMETERS = ["updater_name", "updater_email"]` first and remove the else statement.

**Review by ghost — CHANGES_REQUESTED** [review]

File: tools/upgrade_versions.py (hunk 9 above)
Comment (alanoe) [reply tools/upgrade_versions.py]: Done

**Review by ghost — APPROVED** [review]
